l10n_th_fields/models/partner.py

```diff
@@ -61,11 +61,8 @@ def _check_partner_name(self):
     @api.constrains('vat', 'taxbranch')
     def _check_vat_taxbranch_unique(self):
         if self.vat or self.taxbranch:
-            count = self.search_count(
-                ['|', ('parent_id', '=', False),
-                 ('is_company', '=', True),
-                 ('vat', '=', self.vat),
-                 ('taxbranch', '=', self.taxbranch)])
+            count = self.search_count([('vat', '=', self.vat),
+                                       ('taxbranch', '=', self.taxbranch)])
```

Comment:
> Can you tell me what is it for?

We had a discussion in mid August and agreed with this change.  Please find the email thread titled "Re: CFT - constraint on Tax ID and Tax Branch ID".

> Note: I was having problem when I pull the lastest cft code last week. Got warning saving a customer, even it has unique vat + taxbranch.

Are you sure that vat + tax branch was unique?  I get the error in my test env now after pulling your branch because there are customers with parent (company) - child (contact) relationship and they share the same vat (when you create a contact for a company, by default vat is inherited from the parent, which is Odoo's standard design).  So, the whole point of my above adjustment is to bypass these cases in checking.